create_driver/src/create_driver.cpp

```diff
@@ -31,22 +31,31 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include <string>
 
-CreateDriver::CreateDriver(ros::NodeHandle& nh)
+
+CreateDriver::CreateDriver(std::shared_ptr<rclcpp::Node> nh)
   : model_(create::RobotModel::CREATE_2),
     nh_(nh),
-    priv_nh_("~"),
-    diagnostics_(),
+    priv_nh_(rclcpp::Node::make_shared("_")),
+    tf_broadcaster_(nh),
+    diagnostics_(nh),
     is_running_slowly_(false)
 {
-  std::string robot_model_name;
-  priv_nh_.param<std::string>("dev", dev_, "/dev/ttyUSB0");
-  priv_nh_.param<std::string>("robot_model", robot_model_name, "CREATE_2");
-  priv_nh_.param<std::string>("base_frame", base_frame_, "base_footprint");
-  priv_nh_.param<std::string>("odom_frame", odom_frame_, "odom");
-  priv_nh_.param<double>("latch_cmd_duration", latch_duration_, 0.2);
-  priv_nh_.param<double>("loop_hz", loop_hz_, 10.0);
-  priv_nh_.param<bool>("publish_tf", publish_tf_, true);
-
+  priv_nh_->declare_parameter<std::string>("dev", "/dev/ttyUSB0");
+  priv_nh_->declare_parameter<std::string>("robot_model", "CREATE_2");
+  priv_nh_->declare_parameter<std::string>("base_frame", "base_footprint");
+  priv_nh_->declare_parameter<std::string>("odom_frame", "odom");
+  priv_nh_->declare_parameter<double>("latch_cmd_duration", 0.2);
+  priv_nh_->declare_parameter<double>("loop_hz", 10.0);
+  priv_nh_->declare_parameter<bool>("publish_tf", true);
+
+  priv_nh_->get_parameter("dev", dev_);
```

Comment:
nitpick: we can effectively get the value from the return of `declare_parameter` above. For example:

```c
dev_ = nh_->declare_parameter<std::string>("dev", "/dev/ttyUSB0");
```

This will make the code a bit more compact.

Reply:
Good catch -- I'm not very familiar with this interface. Done.